Import auto-scroll update from Icon-O-Matic that use system auto-scroll

jscipione opened the pull request:
In [hrev57439](https://git.haiku-os.org/haiku/tag/?h=hrev57439) auto-scroll was re-introduced on BListView's. Unfortunately this conflicts with the auto-scroll feature in Icon-O-Matic and WonderBrush which are using GetMouse() to do their own auto-scroll. So instead, I have taken the auto-scroll feature out of Icon-O-Matic here https://review.haiku-os.org/c/haiku/+/7215 and would like to do the same in WonderBrush. This alters WonderBrush\ to use the BListView version of auto-scroll instead.

jscipione committed:
WonderBrush: Import Icon-O-Matic Listviews, add dependencies, fixup

jscipione committed:
WonderBrush: Fix crash on quit deleting BubbleHelper

stippi commented:
Hi! Thanks for the patch. To be honest, I don't think it is good to make changes in Haiku which /require/ patches to applications to make them work again. Instead, the changes should be compatible and applications should be allowed to "opt in". From what I understand, Icon-O-Matic was now crashing and instead of fixing libbe, Icon-O-Matic was changed instead. Please correct me if I misunderstood things.

humdingerb commented:
FWIW, this patch fails to build:

```
WonderBrush/src/gui/listviews/ListViews.cpp:781:1: error: default argument given for parameter 2 of 'status_t SimpleListView::ArchiveSelection(BMessage*, bool) const' [-fpermissive]
  781 | SimpleListView::ArchiveSelection(BMessage*, bool = true) const
      | ^~~~~~~~~~~~~~
In file included from WonderBrush/src/gui/listviews/ListViews.cpp:20:
WonderBrush/src/gui/listviews/ListViews.h:157:41: note: previous specification in 'virtual status_t SimpleListView::ArchiveSelection(BMessage*, bool) const' here
  157 |         virtual status_t                ArchiveSelection(BMessage*, bool = true) const;
      |                                         ^~~~~~~~~~~~~~~~

```

jscipione commented:
Fixed in upstream in hrev57966. The similar fix I made in Icon-O-Matic can stay as it fixes some other issues like redrawing the drop indicator correctly. However that's not a problem on WonderBrush and the crash is now fixed for R1B5.
